### PR TITLE
If Button has no text, require ariaLabel or ariaLabelledBy

### PR DESCRIPTION
**Pull request by jordankoschei-okta**
Does what it says on the box.

### CODE REVIEW

**Comment by edburyenegren-okta** [comment]
If we're already inserting a `Tooltip` around icon-only `Button`s, shouldn't this be automatic?

**Comment by jordankoschei-okta** [comment]
> If we're already inserting a `Tooltip` around icon-only `Button`s, shouldn't this be automatic?

Oh, that's a good call — @KevinGhadyani-Okta , what do you think?

**Comment by jordankoschei-okta** [comment]
**Note for review with KG:** Shred proposed that we don't need this because icon-only buttons should have a `Tooltip`. It might be nice to offer this as an escape hatch, but I totally see the point. What do you think?

File: packages/odyssey-react-mui/src/Button.tsx [hunk 1]
```diff
@@ -79,7 +79,23 @@ export type ButtonProps = {
...
+      text?: undefined | "";
+      ariaLabel: string;
+      ariaLabelledBy?: string;
+    }
+  | {
+      text?: undefined | "";
+      ariaLabel?: string;
+      ariaLabelledBy: string;
+    }
+);
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/Button.tsx:98]: Ask Shred about situations where you have an icon-only button but no tooltip

Comment (edburyenegren-okta) [reply packages/odyssey-react-mui/src/Button.tsx:98]: Our current guidance/intent is for all icon-only Buttons to be tooltipped. That way they're both consistent and provide an affordance. `aria-` attributes aren't as useful for users who aren't on an AST but still need context.

**Comment by jordankoschei-okta** [comment]
Closing this PR with the understanding that we don't need to require these ARIA attributes since an icon-only button should include a tooltip.

**Pull request closed without merging**